src/api/config.c

```diff
@@ -612,6 +612,49 @@ m64p_error ConfigShutdown(void)
 /* Selector functions, exported outside of the Core */
 /* ------------------------------------------------ */
 
+EXPORT m64p_error CALL ConfigExternalGetParameter(const char *FileName, const char *SectionName, const char *ParamName, char* ParamPtr)
+{
+    FILE *fPtr;
+    char buffer[256];
+    int foundSection = 0;
+    if (ParamPtr == NULL || SectionName == NULL || ParamName == NULL)
+        return M64ERR_INPUT_INVALID;
+    if (FileName == NULL || (fPtr = fopen(FileName, "rb")) == NULL)
+    {
+        DebugMessage(M64MSG_ERROR, "Unable to open config file '%s'.", FileName);
+        return M64ERR_INPUT_INVALID;
+    }
+    while (fgets(buffer, 255, fPtr) != NULL)
+    {
+        char *line = buffer;
+        ini_line l = ini_parse_line(&line);
+        switch (l.type)
+        {
+            case INI_SECTION:
+                if (osal_insensitive_strcmp(SectionName, l.name) == 0)
+                    foundSection = 1;
+                else
+                    foundSection = 0;
+                break;
+            case INI_PROPERTY:
+                if (foundSection)
+                {
+                    if (osal_insensitive_strcmp(ParamName, l.name) == 0)
+                    {
+                        fclose(fPtr);
+                        strncpy(ParamPtr, l.value, 256);
```

Comment:
Developers who don't read the documentation are not very good. ;)  I agree in principle that a memory leak is better than memory corruption.  Apple's Cocoa API has a convention in the method names whereby it's clear whether or not the caller is expected to destroy an object that gets returned, but we have no such convention.  Using a static buffer to return to the caller is an okay solution but it's not thread-safe.  If the front-end application was using threads and calling this function at the same time as a plugin, it could cause trouble.  I think the best overall solution is to add an extra parameter, ParamMaxLength or something like that, to specify the size of the buffer being passed in by the caller.